doc/scripts/filter-doc-log.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+# run the filter-known-issues.py script to remove "expected" warning
+# messages from the output of the document build process
+# Only argument is the name of the log file saved by the build.
+
+$ZEPHYR_BASE/scripts/filter-known-issues.py --config-dir \
+      $ZEPHYR_BASE/.known-issues/doc $1 > $1.warn || \
+      (echo " ==> Error running filter-known-issues"; exit 1); \
+      if [ -s $1.warn ]; then \
+        echo " => New documentation warnings/errors"; \
+        cat $1.warn;
+        exit 1; \
+      fi;
```

Comment:
this is unreadable, we are not in a Makefile here, something like this would be better (not tested):

```
KI_SCRIPT=${ZEPHYR_BASE}/scripts/filter-known-issues.py
CONFIG_DIR=${ZEPHYR_BASE}/.known-issues/doc

OUTPUT_FILE=$1
WARNING_FILE=${OUTPUT_FILE}.warn

if [ -z "${WARNING_FILE}" ]; then
        echo "missing parameter"
        exit 1
fi

$KI_SCRIPT --config-dir ${CONFIG_DIR} ${OUTPUT_FILE} > ${WARNING_FILE}
if [ "$?" != 0 ]; then
        echo " ==> Error running filter-known-issues"
        exit 1
fi
if [ -s ${WARNING_FILE} ]; then
        echo " => New documentation warnings/errors"
        cat ${WARNING_FILE}
        exit 1
fi
rm -f ${WARNING_FILE}
rm -f ${OUTPUT_FILE}
```

Reply:
since the check-compliance script expects to find the doc.warnings file, shouldn't delete it here :)  I'll rewrite all this and submit anew.

Reply:
you do need to delete it somehow, otherwise /tmp/ would be filled with junk